Android/Samples/libraries/kml/src/main/java/com/microsoft/maps/kml/KMLParser.java

```diff
@@ -196,15 +204,62 @@ private void parseLineStyle(@NonNull StylesHolder stylesHolder)
                   + "Instead saw int value: "
                   + width);
         }
-        stylesHolder.getLineStyle().setWidth(width);
+        lineStyle.setWidth(width);
       } else if (type.equals("color")) {
-        stylesHolder.getLineStyle().setStrokeColor(parseColor());
+        lineStyle.setStrokeColor(parseColor());
       } else {
         skipToEndOfTag();
       }
     }
   }
 
+  private void parsePolyStyle(@NonNull PolyStyle polyStyle)
+      throws XmlPullParserException, IOException, KMLParseException {
+    while (moveToNext() != XmlPullParser.END_TAG) {
+      if (mParser.getEventType() != XmlPullParser.START_TAG) {
+        continue;
+      }
+      switch (mParser.getName()) {
+        case "fill":
+          String shouldFill = parseText();
+          if (shouldFill.equals("1") || shouldFill.equalsIgnoreCase("true")) {
+            polyStyle.setShouldFill(true);
+          } else if (shouldFill.equals("0") || shouldFill.equalsIgnoreCase("false")) {
+            polyStyle.setShouldFill(false);
+          } else {
+            throw new KMLParseException(
+                "Invalid value ("
+                    + shouldFill
+                    + ") for <fill> at position "
+                    + mParser.getPositionDescription());
+          }
+          break;
+        case "outline":
+          String shouldOutline = parseText();
+          if (shouldOutline.equals("1") || shouldOutline.equalsIgnoreCase("true")) {
+            polyStyle.setShouldOutline(true);
+          } else if (shouldOutline.equals("0") || shouldOutline.equalsIgnoreCase("false")) {
+            polyStyle.setShouldOutline(false);
+          } else {
+            throw new KMLParseException(
+                "Invalid value ("
+                    + shouldOutline
+                    + ") for <outline> at position "
+                    + mParser.getPositionDescription());
+          }
+          break;
+        case "color":
+          if (polyStyle.shouldFill()) {
+            polyStyle.setFillColor(parseColor());
```

Comment:
This seems wrong. What if fill appears after the color tag?

Reply:
Changed this to always set fillColor to what is specified while parsing. When applying the styles, the fillColor of the polygon is only set if polyStyle.shouldFill() is true (else fillColor is set to transparent).